Enterprise-CMCS/managed-care-review
# MCR-3940 Change RateDetails page to call fetchContract

## Summary
Main work completed:
- Implement  `MULTI` RateDetails workflow
   - I saw some rendering issues after adding additional `useQuery` calls for the new APIs. I spent the most time on trying to investigate what is going on with re-renders.
- Change the API that is called to be conditional -  contract API if we are in MULTI, rate API if we are in SINGLE 

Other notes:
- noticed there's still something odd going on with `ProgramSelect` in terms of that UI triggering additional re-renders. However, addressing that seemed out of scope + needs more investigation. It is an existing issue.

#### Related issues
https://jiraent.cms.gov/browse/MCR-3940

#### Screenshots

UI with linked rate flag off (red box shows the packages with shared rates UI is present) 
![Screenshot 2024-03-04 at 5 52 02 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/10750442/e9da789e-3289-488c-ae99-761bbd40ecdd)

UI with linked rate flag on (no red box - these fields are not visible and we should not validate on them)
![Screenshot 2024-03-04 at 5 52 53 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/10750442/81a0e637-eece-4f81-8e8a-78f3af9718eb)


#### Test cases covered
- I'm starting to write Jest tests. Will keep incrementally committing these.  Let's not hold up on the PR on tests though -   this is still featured flagged (in use on rate edit page and on new linked rate workflow). 
- Cypress must pass, that helps ensure I didn't change flag off behavior



## QA guidance
- **Open Network tab** - you should see `/graphql` request to fetch contract when the linked rates feature flag is on. 
- **Look at the UI** - There is no `PackagesWithSharedRates` UI shown when linked rates feature flag is on. Validations still work (e.g. no validation error for "You must select at least one submission" when you continue with no changes)
-  **Cypress** - Existing cypress tests for linked rates flag OFF still pass
